src/Haskell/Ide/Engine/Options.hs

```diff
@@ -14,40 +14,66 @@ import           System.IO
 import qualified System.Log.Logger                     as L
 import           Data.Foldable
 
+data ProjectLoadingOpts = ProjectLoadingOpts
+  { optDryRun        :: Bool
+  , optFiles         :: [FilePath]
+  } deriving (Show, Eq)
+
+data RunMode = LspMode | ProjectLoadingMode ProjectLoadingOpts
+  deriving (Show, Eq)
+
 data GlobalOpts = GlobalOpts
   { optDebugOn       :: Bool
   , optLogFile       :: Maybe String
-  , optLsp           :: Bool
   , projectRoot      :: Maybe String
   , optBiosVerbose   :: Bool
   , optCaptureFile   :: Maybe FilePath
   , optExamplePlugin :: Bool
-  , optDryRun        :: Bool
-  , optFiles         :: [FilePath]
-  } deriving (Show)
+  , optMode          :: RunMode
+  } deriving (Show, Eq)
 
 -- | Introduced as the common prefix of app/HieWrapper.hs/main and app/MainHie.hs/main
 initApp :: String -> IO GlobalOpts
 initApp namedesc = do
   hSetBuffering stderr LineBuffering
-  let numericVersion :: Parser (a -> a)
-      numericVersion = infoOption (showVersion Meta.version)
-        (long "numeric-version" <> help "Show only version number")
-      compiler :: Parser (a -> a)
-      compiler = infoOption hieGhcDisplayVersion
-        (long "compiler" <> help "Show only compiler and version supported")
     -- Parse the options and run
   (opts, ()) <- simpleOptions
     hieVersion
     namedesc
     ""
-    (numericVersion <*> compiler <*> globalOptsParser)
+    optionParser
     empty             
   Core.setupLogger (optLogFile opts) ["hie", "hie-bios"]
     $ if optDebugOn opts then L.DEBUG else L.INFO
   traverse_ setCurrentDirectory $ projectRoot opts
   return opts
 
+optionParser :: Parser GlobalOpts
+optionParser = numericVersion <*> compiler <*> globalOptsParser
+
+numericVersion :: Parser (a -> a)
+numericVersion = infoOption (showVersion Meta.version)
+  (long "numeric-version" <> help "Show only version number")
+
+compiler :: Parser (a -> a)
+compiler = infoOption hieGhcDisplayVersion
+  (long "compiler" <> help "Show only compiler and version supported")
+
+projectLoadingModeParser :: Parser RunMode
+projectLoadingModeParser =
+  ProjectLoadingMode
+  <$> (ProjectLoadingOpts
+       <$> flag False True
+          (  long "dry-run"
+            <> help "Perform a dry-run of loading files. Only searches for Haskell source files to load. Does nothing if run as LSP server."
```

Comment:
Line too long, limiting yourself to 80 chars per line makes it easier to read on small monitors